rust-tide-backend/src/app.rs

```diff
@@ -0,0 +1,44 @@
+use crate::{error::Error::*, handler, settings::logging, Result, CONFIG};
+use log::info;
+
+pub struct AppState {
+    pub sample_state: String,
+}
+
+impl AppState {
+    pub fn new() -> Self {
+        AppState {
+            sample_state: String::from("dummy state"),
+        }
+    }
+}
+
+pub struct App {}
+
+impl App {
+    pub fn new() -> Result<App> {
+        logging::init(&CONFIG.log.level);
+        Ok(App {})
+    }
+
+    pub async fn run(&self) -> Result<()> {
+        info!("Starting with environment: {}", CONFIG.env);
+
+        let app_state = AppState::new();
+
+        let addr: String = format!("0.0.0.0:{}", CONFIG.server.port).parse().unwrap();
```

Comment:
```suggestion
        let addr: String = format!("0.0.0.0:{}", CONFIG.server.port).parse().expect("Couldn't parse server port");
```
not sure if `unwrap` gives a nice error message here....

Reply:
good point 👍